src/main/java/gregtech/common/covers/CoverFluidVoiding.java

```diff
@@ -0,0 +1,148 @@
+import codechicken.lib.render.pipeline.IVertexOperation;
+import codechicken.lib.vec.Cuboid6;
+import codechicken.lib.vec.Matrix4;
+import gregtech.api.capability.GregtechCapabilities;
+import gregtech.api.capability.IFluidVoiding;
+import gregtech.api.capability.impl.FluidHandlerDelegate;
+import gregtech.api.cover.CoverBehavior;
+import gregtech.api.cover.CoverWithUI;
+import gregtech.api.cover.ICoverable;
+import gregtech.api.gui.GuiTextures;
+import gregtech.api.gui.ModularUI;
+import gregtech.api.gui.widgets.LabelWidget;
+import gregtech.api.gui.widgets.WidgetGroup;
+import gregtech.api.metatileentity.MetaTileEntity;
+import gregtech.api.render.Textures;
+import gregtech.common.covers.filter.FluidFilterContainer;
+import net.minecraft.block.Block;
+import net.minecraft.entity.player.EntityPlayer;
+import net.minecraft.entity.player.EntityPlayerMP;
+import net.minecraft.item.ItemStack;
+import net.minecraft.nbt.NBTTagCompound;
+import net.minecraft.util.*;
+import net.minecraftforge.common.capabilities.Capability;
+import net.minecraftforge.fluids.FluidStack;
+import net.minecraftforge.fluids.capability.CapabilityFluidHandler;
+import net.minecraftforge.fluids.capability.IFluidHandler;
+
+import java.util.function.Predicate;
+
+public class CoverFluidVoiding extends CoverBehavior implements CoverWithUI, IFluidVoiding {
+
+    private FluidFilterContainer fluidFilter;
+    protected FluidHandlerVoiding fluidHandler;
+    private final int voidingAmount = Integer.MAX_VALUE;
+
+    public CoverFluidVoiding(ICoverable coverable, EnumFacing attachedSide) {
+        super(coverable, attachedSide);
+        fluidFilter = new FluidFilterContainer(this);
+    }
+
+    @Override
+    public int voidingAmount() {
+        return voidingAmount;
+    }
+
+    @Override
+    public boolean canAttach() {
+        return coverHolder.getCapability(CapabilityFluidHandler.FLUID_HANDLER_CAPABILITY, attachedSide) != null;
+    }
+
+    @Override
+    public void renderCover(CCRenderState renderState, Matrix4 translation, IVertexOperation[] pipeline, Cuboid6 plateBox, BlockRenderLayer layer) {
+        Textures.PUMP_OVERLAY.renderSided(attachedSide, plateBox, renderState, pipeline, translation);
+    }
+
+    @Override
+    public void onRemoved() {
+        NonNullList<ItemStack> drops = NonNullList.create();
+        MetaTileEntity.clearInventory(drops, fluidFilter.getFilterInventory());
+        for (ItemStack itemStack : drops) {
+            Block.spawnAsEntity(coverHolder.getWorld(), coverHolder.getPos(), itemStack);
+        }
+    }
+
+    @Override
+    public EnumActionResult onScrewdriverClick(EntityPlayer playerIn, EnumHand hand, CuboidRayTraceResult hitResult) {
+        if (!coverHolder.getWorld().isRemote) {
+            openUI((EntityPlayerMP) playerIn);
+        }
+        return EnumActionResult.SUCCESS;
+    }
+
+    @Override
+    public boolean shouldCoverInteractWithOutputside() {
+        return true;
+    }
```

Comment:
This should be false as it is only for covers which could input from output side and automatically allow that input. But name of function is not descriptive...